akka-docs/rst/scala/distributed-data.rst

```diff
@@ -0,0 +1,499 @@
+  at least **N/2 + 1** replicas, where N is the number of nodes in the cluster
+  (or cluster role group)
+* ``ReadAll`` the value will be read and merged from all nodes in the cluster
+  (or all nodes in the cluster role group)
+
+
+.. includecode:: code/docs/ddata/DistributedDataDocSpec.scala#get
+
+As reply of the ``Get`` a ``Replicator.GetSuccess`` is sent to the sender of the
+``Get`` if the value was successfully retrieved according to the supplied consistency
+level within the supplied timeout. Otherwise a ``Replicator.GetFailure`` is sent.
+If the key does not exist the reply will be ``Replicator.NotFound``.
+
+.. includecode:: code/docs/ddata/DistributedDataDocSpec.scala#get-response1
+
+.. includecode:: code/docs/ddata/DistributedDataDocSpec.scala#get-response2
+
+You will always read your own writes. For example if you send a ``Update`` message
+followed by a ``Get`` of the same ``key`` the ``Get`` will retrieve the change that was
+performed by the preceding ``Update`` message. However, the order of the reply messages are
+not defined, i.e. in the previous example you may receive the ``GetSuccess`` before
+the ``UpdateSuccess``.
+
+In the ``Get`` message you can pass an optional request context in the same way as for the
+``Update`` message, described above. For example the original sender can be passed and replied
+to after receiving and transforming ``GetSuccess``.
+
+.. includecode:: code/docs/ddata/DistributedDataDocSpec.scala#get-request-context
+
+You can retrieve all keys of a local replica by sending ``Replicator.GetKeys`` message to the
+``Replicator``. The reply of ``GetKeys`` is a ``Replicator.GetKeysResult`` message.
+
+Consistency
+-----------
+
+The consistency level that is supplied in the :ref:`replicator_update` and :ref:`replicator_get`
+specifies per request how many replicas that must respond successfully to a write and read request.
+
+For low latency reads you use ``ReadLocal`` with the risk of retrieving stale data, i.e. updates
+from other nodes might not be visible yet. 
+
+When using ``WriteLocal`` the update is only written to the local replica and then disseminated
+in the background with the gossip protocol, which can take few seconds to spread to all nodes.
+
+``WriteAll`` and ``ReadAll`` is the strongest consistency level, but also the slowest and with
+lowest availability. For example, it is enough that one node is unavailable for a ``Get`` request
+and you will not receive the value.
+
+If consistency is important, you can ensure that a read always reflects the most recent
+write by using the following formula::
+
+    (nodes_written + nodes_read) > N 
+
+where N is the total number of nodes in the cluster, or the number of nodes with the role that is
+used for the ``Replicator``.
+
+For example, in a 7 node cluster you get this strong read consistency by writing to 4 nodes and
+reading from 4 nodes, or writing to 5 nodes and reading from 3 nodes.
```

Comment:
Good point, I was trying to describe "ensure that a read always reflects the most recent write" (strong read consistency). I agree that we should reword to avoid confusion.